02-csv-vector-raster-plotting.Rmd

```diff
@@ -115,107 +118,133 @@ head(plot.location)
 
 ```
 
-When we create a spatial points dataframe from a csv, there is no inherent `CRS` 
-associated with it. We can assign a CRS if we know what the coordinate system 
-the x,y values are in.
-
-In this case, the CRS of our csv is UTM zone 18N (similar to our boundary shapefiles) 
-that we opened previously. If the object is not still open in R, let's open up a 
-shapefile that is in that project. Next, we will convert our imported `plot.location` 
-object to a `SpatialPointsDataFrame` and assign it a CRS using the shapefile.
+#Challenge: Reading data structure
+ * What is the coordinate reference system (CRS) for this data? 
+ 
 
+When we create a spatial points dataframe from a csv, there is no inherent
+Coordinate Reference System (CRS) associated with it. We can assign a CRS if we
+know what the coordinate system the x,y values are in.
 
-NOTE: it is important to first CONFIRM that the CRS of our CSV coordinates is the 
-SAME as the CRS of the shapefile. We can figure that out by looking at the metadata 
-for the CSV file which we hopefully have!
+In this case, the CRS for this data is UTM zone 18N (utmZone is 18N for all
+points in the header we just looked at). Luckily this is similar to our boundary
+shapefiles that we opened previously. 
 
-#NOTE: create metadata for this file!
+If it is not still open in R, let's open up the shapefile `HarClip_UTMZ18".
 
 ```{r read-shp }
-#note: read ogr is preferred as it maintains prj info
+
 aoiBoundary <- readOGR("boundaryFiles/HARV/","HarClip_UTMZ18")
+```
 
-#check the CRS
-crs(aoiBoundary)
+This is the shapefile that we want to match up with our csv file.  It is
+important to first confirm that the CRS of our CSV coordinates is the same as 
+the CRS of the shapefile. We can figure that out by looking at the metadata for
+both files.
 
-#the x,y location values for the dataset are available in 
-#the x and y columns. We can use that for our spatial data point object
-#we know the csv file coordinates are in the same CRS as the other shapefiles.
-#we can assign this to the SpatialPointsDataFrame when we import it.
-plot.locationSp <- SpatialPointsDataFrame(plot.location[,1:2],
-                                                plot.location,
-                                                proj4string = crs(aoiBoundary))
+```{r check-crs-shapefile-csv}
+#check the CRS in shapefile
+crs(aoiBoundary)
+```
 
+Next, we will convert our imported `plot.location` object to a
+`SpatialPointsDataFrame` and assign it a CRS using the shapefile.
+
+The x,y location values for the dataset are available and named "x" (column 1) and
+"y" (column 2).  We can use that for our spatial data point objects. We know the csv file 
+coordinates are in the same CRS as the other shapefiles. We can assign this to the 
+`SpatialPointsDataFrame` when we import it.
+
+``` {r plot-csv-shapefile}
+#
+plot.locationSp <- SpatialPointsDataFrame(plot.location[,1:2],    #x,y coordinates
```

Comment:
plot.LocationSp <- 
        SpatialPointsDataFrame(plot.location[,1:2],    #the columns which represent x,y coordinate locations 
         plot.location,  #the R object to convert to a SPDataFrame
         proj4string = crs(aoiBoundary)) #the CRS to assign to the SPDataFrame - in this instance we are pulling the CRS from the aoiBoundary object